card/lib/generators/card/card_generator.rb

```diff
@@ -0,0 +1,33 @@
+# -*- encoding : utf-8 -*-
+
+class Card
+  module Generators
+    class CardGenerator < NamedBase
+      include Rails::Generators::Migration
+      extend ActiveRecord::Generators::Migration
+
+      source_root File.expand_path('../templates', __FILE__)
+
+      argument :mount_point, :required=>false
+      argument :prefix, :required=>false
+
+      def self.namespace
+        'card'
+      end
+
+      def create_files
+        name_prefix = name == 'base' ? '' : name+'_'
+        template 'config/routes.erb', "config/#{name_prefix}routes.rb"
+
+        mig_name = 'create_cards_tables'
+        dirname = 'db/migrations'
+        if destination = self.class.migration_exists?(dirname, mig_name) and options.force?
+          remove_file(destination)
+        elsif destination
+          raise "Another migration is already named #{mig_name}: #{destination}"
+        end
```

Comment:
This code is stolen from the rails generators because I couldn't figure out how to make a generator class that had both a migration with those filename patterns and also do some other templates.